circuits-circom/hdfc_registration.circom

```diff
@@ -0,0 +1,86 @@
+    var max_email_to_len = ceil(35, pack_size); // TODO: Change this.
+    var max_account_number_len = ceil(7, pack_size);    // TODO: Change this. Max is only 4 digits but set to 7 for now.
+
+    signal input in_padded[max_header_bytes]; // prehashed email data, includes up to 512 + 64? bytes of padding pre SHA256, and padded with lots of 0s at end after the length
+    signal input modulus[k]; // rsa pubkey, verified with smart contract + DNSSEC proof. split up into k parts of n bits each.
+    signal input signature[k]; // rsa signature. split up into k parts of n bits each.
+    signal input in_len_padded_bytes; // length of in email data including the padding, which will inform the sha256 block length
+
+    // Base 64 body hash variables
+    signal input body_hash_idx;
+    // The precomputed_sha value is the Merkle-Damgard state of our SHA hash uptil our first regex match which allows us to save SHA constraints by only hashing the relevant part of the body
+    signal input precomputed_sha[32];
+    // Suffix of the body after precomputed SHA
+    signal input in_body_padded[max_body_bytes];
+    // Length of the body after precomputed SHA
+    signal input in_body_len_padded_bytes;
+
+    signal output modulus_hash;
+
+    // DKIM VERIFICATION
+    component EV = EmailVerifier(max_header_bytes, max_body_bytes, n, k, 0);
+    EV.in_padded <== in_padded;
+    EV.pubkey <== modulus;
+    EV.signature <== signature;
+    EV.in_len_padded_bytes <== in_len_padded_bytes;
+    EV.body_hash_idx <== body_hash_idx;
+    EV.precomputed_sha <== precomputed_sha;
+    EV.in_body_padded <== in_body_padded;
+    EV.in_body_len_padded_bytes <== in_body_len_padded_bytes;
+    signal header_hash[256] <== EV.sha;
+
+    modulus_hash <== EV.pubkey_hash;
+
+    // HDFC UPI EMAIL VERIFICATION REGEX (Check that regex matches)
+    signal upi_subject_regex_out <== HdfcUpiSubjectRegex(max_header_bytes)(in_padded);
+    upi_subject_regex_out === 1;
+
+    // FROM HEADER REGEX
+    var max_email_from_packed_bytes = count_packed(max_email_from_len, pack_size);
+    assert(max_email_from_packed_bytes < max_header_bytes);
+    signal input email_from_idx;
+    signal output reveal_email_from_packed[max_email_from_packed_bytes]; // packed into 7-bytes
+    signal (from_regex_out, from_regex_reveal[max_header_bytes]) <== FromRegex(max_header_bytes)(in_padded);
+    from_regex_out === 1;
+    reveal_email_from_packed <== ShiftAndPackMaskedStr(max_header_bytes, max_email_from_len, pack_size)(from_regex_reveal, email_from_idx);
+
+    // Extract packed and hashed onramper id
+    signal input email_to_idx;
+    signal input hdfc_acc_num_idx;
+    signal output onramper_id <== HdfcOnramperId(
+        max_email_to_len, 
+        max_header_bytes, 
+        max_account_number_len, 
+        max_body_bytes, 
+        pack_size
+    )(in_padded, email_to_idx, in_body_padded, hdfc_acc_num_idx);
+
+    // TOTAL CONSTRAINTS: (WITH STUB): 936240
+    // WITHOUT STUB: TODO
```

Comment:
Let's recompile with the stub. Any reason not to?

Reply:
Not necessary while we are still in dev and iterating on circuits.